packages/eslint-plugin/tests/worksWithStringlyTypedExtends.test.ts

```diff
@@ -0,0 +1,25 @@
+import { defineConfig } from 'eslint/config';
+
+import plugin from '../src/index';
+
+describe("The plugin object should work with eslint defineConfig's stringly typed extends array", () => {
+  it('should not error when explicitly referencing the flat version', () => {
+    defineConfig({
+      extends: ['ts/flat/strict'],
+      plugins: {
+        // @ts-expect-error -- types aren't compatible.
```

Comment:
side note:
it's been on my todo list for a while -- we need to just bite the bullet and make them compatible.